iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/ArmBaseConverter.java

```diff
@@ -100,19 +100,39 @@ public ObjectExpression convertToObjectExpression(MappingTree tree) {
   public ArrayExpression convertToArrayExpression(SequenceTree tree) {
     return new ArrayExpressionImpl(tree.metadata(),
       tree.elements().stream()
-        .map(this::convertToPropertyValue)
+        .map(this::convertToExpression)
         .collect(Collectors.toList()));
   }
 
-  public PropertyValue convertToPropertyValue(YamlTree tree) {
+  public Expression convertToExpression(YamlTree tree) {
     if (tree instanceof SequenceTree) {
       return convertToArrayExpression((SequenceTree) tree);
     } else if (tree instanceof MappingTree) {
       return convertToObjectExpression((MappingTree) tree);
     } else if (tree instanceof ScalarTree) {
-      return convertToExpression((ScalarTree) tree);
+      return convertToExpressionLiteral((ScalarTree) tree);
     } else {
-      throw new ParseException("Couldn't convert to PropertyValue, unsupported class " + tree.getClass().getSimpleName(), new BasicTextPointer(tree.metadata().textRange()), null);
+      throw new ParseException("Couldn't convert to Expression, unsupported class " + tree.getClass().getSimpleName(), new BasicTextPointer(tree.metadata().textRange()), null);
+    }
+  }
+
+  public Expression convertToExpressionLiteral(ScalarTree tree) {
```

Comment:
```suggestion
  public Expression convertToLiteralExpression(ScalarTree tree) {
```
To be uniform with the other methods let's switch the words.